File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.java
class Solution {
    public String solution(int[] food) {
        // 1, 3, 4, 6 -> "1223330333221"
        StringBuilder sb = new StringBuilder();
        for (int i = 1; i < food.length; i++) {
            for (int j = 0; j < food[i] / 2; j++) {
                sb.append(String.valueOf(i));
            }
        }

        sb.append("0");

        for (int i = food.length - 1; i > 0; i--) {
            for (int j = 0; j < food[i] / 2; j++) {
                sb.append(String.valueOf(i));
            }
        }

        return sb.toString();
    }
}

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.py
def solution(food):
    res = []
    for i in range(1, len(food)):
        for j in range(food[i] // 2):
            res.append(str(i))    
    tmp = ''.join(res[::-1])
    res.append('0')
    res.append(tmp)
    return ''.join(res)
    